### DISCUSSION TITLE
A question regarding code style

### DISCUSSION THREAD

**Discussion by DanielMowitz** [GothenburgBitFactory/taskwarrior#2446] [General]
Browsing the code, I noticed that the pattern `Context::getContext` is used very often (917 times according to a quick grep). Many times the context getter is called just to retrieve one value, which happens multiple times per method. This could be shortened by instead using `auto& context = Context::getContext ();` once and then just using `context`.

I am aware that the [coding style page](https://taskwarrior.org/docs/coding_style.html) states: 

> Make all changes and additions such that they blend in perfectly with the surrounding code, so it looks like only one person worked on the source, and that person is rigidly consistent.

Yet I'd like to ask the imaginary single maintainer to overthink their coding style a little ;)
If this is wanted I'd be more than happy to go through the source and apply this change everywhere.

**Comment by tbabej** [comment]
Hi @DanielMowitz,

I did not author the guidelines, but I understand them, they are talking more about formatting than usage of `Context::getContext ()`. I would not mind merging such a patch, at the same time, if you want to help, I'd argue there are more pressing issues in the codebase! Either way, I'm happy to hop on #taskwarrior and discuss.
